### DISCUSSION TITLE
Uptime format

### DISCUSSION THREAD

**Discussion by sadata7** [GitSquared/edex-ui#1065] [Questions] [answered]
Is it possible, without changing themes, to configure uptime to display in a more useful format (e.g. 28d:17h:32m)?  If this is not currently possible, can it be added in the future?  The 'd,' 'm,' and 'h' could be displayed in the same color as the labels above the value so that the numbers stand out.  If space becomes an issue, I guess you could remove the POWER ON item, since that is not all that useful.

**Comment by GitSquared** [comment]
We could look into displaying dms instead of hms, i agree its more useful. Regarding the power on thing, it displays current battery % on devices that have one.

**Reply by sadata7** [reply]
It appears that it's already displaying dms but, without the labels, it looks like hms at first glance (of course, the "seconds" aren't constantly changing, so it soon becomes evident that they are not seconds).  Perhaps you could get by just by replacing the first colon with a 'd' (e.g. 28d17:32).  That alone would be a big improvement and you wouldn't lose much real estate.

**Reply by sadata7** [reply]
I made a little hack to sysinfo.class.js. A two- or three-digit day value will cause the date to wrap, but doesn't look bad. A variation (0d00:00) prevents wrapping even with 3-digit days.  See attached screenshots.

`document.querySelector("#mod_sysinfo > div:nth-child(2) > h2").innerHTML = uptime.days + '<span style="opacity:0.5;">d</span>' + uptime.hours + '<span style="opacity:0.5;">:</span>' + uptime.minutes;`

Not sure if you have any predefined classes for the different opacities, so I just hard-coded the styles.

![Selection_202](https://user-images.githubusercontent.com/80186985/110585133-5fca9700-813e-11eb-8c59-4649bd23aaf8.png)
![Selection_205](https://user-images.githubusercontent.com/80186985/110586345-154a1a00-8140-11eb-9d09-568024158872.png)
![Selection_203](https://user-images.githubusercontent.com/80186985/110585145-648f4b00-813e-11eb-9ba1-f03c2a267ee7.png)
![Selection_206](https://user-images.githubusercontent.com/80186985/110586350-1a0ece00-8140-11eb-8f15-47994f2458e2.png)

**Reply by GitSquared** [reply]
ooh that looks fine. Love the `0d00:00` variant, would you like to submit a PR for it?

**Comment by indramardiana** [comment]
can you help .. how much money


Pada tanggal Rab, 10 Mar 2021 13:25, Gaby <notifications@github.com>
menulis:

**Comment by GitSquared** [comment, marked as answer]
Merged in 076a788db92d25cd650a70fdb8dcac4418e1d8d6, PR #1084
